mithril-aggregator/src/dependency_injection/builder.rs

```diff
@@ -273,7 +273,10 @@ impl DependenciesBuilder {
     }
 
     async fn build_stake_store(&mut self) -> Result<Arc<StakePoolStore>> {
-        let stake_pool_store = Arc::new(StakePoolStore::new(self.get_sqlite_connection().await?));
+        let stake_pool_store = Arc::new(StakePoolStore::new(
+            self.get_sqlite_connection().await?,
+            self.configuration.store_retention_limit.map(|l| l as u64),
```

Comment:
Do we want to make sure this value is not in a danger zone? Let's say it's safe with `>2`. WDYT?

Reply:
Done with the `safe_epoch_retention_limit` method on the aggregator `Configuration` type :).